src/Algebra/Graph.hs

```diff
@@ -381,25 +388,31 @@ connect = Connect
 -- 'vertexSet'   . vertices == Set.'Set.fromList'
 -- @
 vertices :: [a] -> Graph a
-vertices = overlays . map vertex
```

Comment:
Any hope (perhaps, just in future) to be able to implement `vertices` as `overlays . map vertex` preserving the goodness? :)

Reply:
Sadly, goodness behave well with composition only when types agree.
`overlays . map vertex` will still be a good consumer of lists, but not a good producer, since `overlays` is only a good consumer of lists, and not of lists _of graphs_.
To say it more clearly, if we abstract constructors for the whole `vertices`, we need to reuse the same abstract constructors in `overlays`.

Reply:
I see. I think you have already explained this to me once :) Could you add it as a comment, perhaps, with a TODO (that we'd like to simplify this to `overlays . map vertex`).